plugins/cue_points/base/lib/kCuePointManager.php

```diff
@@ -97,20 +97,43 @@ protected static function handleConcatAfterMultiClipJobFinished($dbBatchJob, $da
 	 * @param int $offset
 	 * @throws PropelException
 	 */
-	protected static function addChapterCuePoint($partnerId, $sourceEntryId, $destEntryId, $offset)
+	protected static function addChapterCuePoint($partnerId, $sourceEntryId, $destEntryId, $offset, $chapterIndex, $chapterNamePolicy)
 	{
 		$chapter = new ThumbCuePoint();
 		$chapter->setPartnerId($partnerId);
 		$chapter->setEntryId($destEntryId);
 		$chapter->setSubType(ThumbCuePointSubType::CHAPTER);
 		$chapter->setThumbOffset($offset);
 		$chapter->setStartTime($offset);
-		$chapter->setName("$sourceEntryId");
+		$chapterName = self::getChapterNameByPolicy($sourceEntryId, $chapterIndex, $chapterNamePolicy);
+		$chapter->setName("$chapterName");
 		$chapter->setSystemName(self::MULTI_CLIP_CHAPTER_SYSTEM_NAME);
 		$chapter->setStatus(KalturaCuePointStatus::READY);
 		$chapter->save();
 	}
 
+	protected static function getChapterNameByPolicy($sourceEntryId, $chapterIndex, $chapterNamePolicy)
+	{
+		$entry = null;
+		if($chapterNamePolicy != ChapterNamePolicy::NUMERICAL)
+		{
+			$entry = entryPeer::retrieveByPK($sourceEntryId);
+		}
+		switch ($chapterNamePolicy)
```

Comment:
if not set it gets as null and falls to the default case?

Reply:
yes